04-onion-routing.md

```diff
@@ -1407,6 +1438,117 @@ The _origin node_:
   - MAY use the data specified in the various failure types for debugging
   purposes.
 
+## Recommendations for Reputation
+
+Local reputation is used by forwarding nodes to decide whether to endorse a HTLC
+on their outgoing link. Nodes MAY use any metric of their choosing to classify
+a peer as having "good" reputation, though a poor choice of reputation scoring
+metric may affect their downstream reputation.
+
+Peers build reputation by forwarding successful HTLCs that resolve quickly, and
+lose reputation if they endorse failing or slow-resolving HTLCs. Reputation is
+only _negatively_ affected if an endorsed HTLC resolves undesirably, to hold
+nodes accountable for their endorsement signal while still allowing them to 
+forward unendorsed HTLCs that they are not certain about.
+
+HTLC resolution time is assessed relative to a threshold that the node 
+considers to be a reasonable amount of time for a HTLC to resolve: 
+- `resolution_period`: the amount of time a HTLC is allowed to resolve in that 
+  is classified as "good" behavior, expressed in seconds (default: 60 seconds).
```

Comment:
>Resolution time is function of many factors beyond the routing hops, such as the host performance of the routing hops along the payment path. A payer might deliberately select a “pleb-net” payment path going through nodes on shitty raspy to preserve its privacy from high-performance routing hops themselves (or deliberately “holding” the HTLC as it has been proposed in the past https://github.com/lightning/bolts/issues/1008).

We can find a reasonable default, and if someone wants to do a-typical, they can set endorsed = 0

Reply:
>Resolution time is function of many factors beyond the routing hops, such as the host performance of the routing hops along the payment path. A payer might deliberately select a “pleb-net” payment path going through nodes on shitty raspy to preserve its privacy from high-performance routing hops themselves (or deliberately “holding” the HTLC as it has been proposed in the past https://github.com/lightning/bolts/issues/1008).

We can find a reasonable default, and if someone wants to do a-typical, they can set endorsed = 0

Reply:
> We can find a reasonable default, and if someone wants to do a-typical, they can set endorsed = 0

Yes ideally this “resolution time” default could be announced as a TLV extension of `channel_announcement`. That way if a routing hop wanna to increase the throughput of its channel, they can do it in a unilateral fashion.

Reply:
This resolution time is an internal detail, there is no reason to leak it. What would other nodes even do with that information?

Reply:
> This resolution time is an internal detail, there is no reason to leak it. What would other nodes even do with that information?

If you wish to apply a lower resolution time than network default, and therefore apply ranges of upfront fess or reputation algorithms outcomes requirements more stricter, because you have a different subjective appreciation of your liquidity ressources.